pkg/metrics/queue.go

```diff
@@ -58,8 +59,9 @@ type QueueMetrics struct {
 	resourceMetricsLabel *prometheus.GaugeVec
 	// Deprecated - To be removed in 1.7.0. Replaced with queue label Metrics
 	resourceMetricsSubsystem *prometheus.GaugeVec
-	// Track known resource types.
+	// Track known resource types
 	knownResourceTypes map[string]struct{}
+	lock               locking.RWMutex
```

Comment:
This doesn't need to be an RWMutex as we never lock for reads.

Reply:
Good catch, addressed in latest PR, thanks!